lib/clone_kit/rule.rb

```diff
@@ -4,11 +4,20 @@ module CloneKit
   class Rule
     attr_accessor :id_generator
 
+    def initialize(id_generator: nil)
+      @id_generator = id_generator
+    end
+
     def current_operation=(operation)
       @shared_id_map = nil
       @current_operation = operation
     end
 
+    def id_generator=(generator)
+      return if id_generator
```

Comment:
🤔 How come we short-circuit here?  It seems like we shouldn't make this an assignment method if it doesn't behave like one.